api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/rules/impl/LateFrenchImmersionRule.java

```diff
@@ -0,0 +1,43 @@
+package ca.bc.gov.educ.studentdatacollection.api.rules.impl;
+
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationFieldCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationIssueSeverityCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.SdcSchoolCollectionStudentValidationIssueTypeCode;
+import ca.bc.gov.educ.studentdatacollection.api.constants.v1.*;
+import ca.bc.gov.educ.studentdatacollection.api.rules.BaseRule;
+import ca.bc.gov.educ.studentdatacollection.api.service.v1.ValidationRulesService;
+import ca.bc.gov.educ.studentdatacollection.api.struct.SdcStudentSagaData;
+import ca.bc.gov.educ.studentdatacollection.api.struct.v1.SdcSchoolCollectionStudentValidationIssue;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class LateFrenchImmersionRule implements BaseRule {
+    private final ValidationRulesService validationRulesService;
+    public LateFrenchImmersionRule(ValidationRulesService validationRulesService) {
+        this.validationRulesService = validationRulesService;
+    }
+    @Override
+    public boolean shouldExecute(SdcStudentSagaData sdcStudentSagaData) {
+        return CollectionTypeCodes.findByValue(sdcStudentSagaData.getCollectionTypeCode(), sdcStudentSagaData.getSchool().getSchoolCategoryCode()).isPresent() &&
+                StringUtils.isNotEmpty(sdcStudentSagaData.getSdcSchoolCollectionStudent().getEnrolledGradeCode())
```

Comment:
For these long complicated checks, it can make it easier to read/understand if you chunk the pieces out into variables with descriptive names.
`var studentHasEnrolledProgramCodes = StringUtils.isNotEmpty(sdcStudentSagaData.getSdcSchoolCollectionStudent().getEnrolledProgramCodes())
var studentIsNotInGradeHs = !sdcStudentSagaData.getSdcSchoolCollectionStudent().getEnrolledGradeCode().equals("HS")

return studentHasEnrolledProgramCodes && studentIsNotInGradeHs`

I only do this with long checks like these, but I know it's a bit of a personal preference thing

Reply:
Its part of the refactoring that I am doing. I will add order to the rules and move some criteria out of the shouldExecute. This PR is just moving each rule to its own class

Reply:
Eventually, these checks with reduce in size